header.go

```diff
@@ -43,6 +43,61 @@ type Header struct {
 	rawTLVs            []byte
 }
 
+// HeaderProxyFromAddrs creates a new PROXY header from a source and a
+// destination address. The latest protocol version is used.
+func HeaderProxyFromAddrs(sourceAddr, destAddr net.Addr) *Header {
+	h := &Header{
+		Version: 2,
+		Command: PROXY,
+	}
+	switch sourceAddr := sourceAddr.(type) {
+	case *net.TCPAddr:
+		destAddr, ok := destAddr.(*net.TCPAddr)
+		if !ok {
+			break
+		}
+		if len(sourceAddr.IP.To4()) == net.IPv4len {
+			h.TransportProtocol = TCPv4
+		} else if len(sourceAddr.IP) == net.IPv6len {
+			h.TransportProtocol = TCPv6
+		} else {
+			break
```

Comment:
Why not set as `UNSPEC`?

Reply:
`UNSPEC` should be the default value for `TransportProtocol`. I can make that explicit if you prefer.

Reply:
Makes it easier for people using this func.